### PR TITLE
Pre-scan Git SSH hosts

### PR DESCRIPTION
**Pull request by brianhlin**
@mmascher and @jdost321 ran into host verification failures when accessing a Git SSH endpoint

I wish you could chain param expansions...

### CODE REVIEW

File: bosco-override-setup.sh [hunk 1]
```diff
@@ -20,6 +20,12 @@ function errexit {
 GIT_ENDPOINT=$1
 RESOURCE_NAME=$2
 
+# pre-scan the Git repo's host key
+if [[ $GIT_ENDPOINT =~ ^[A-Za-z0-9]+@ ]]; then
```
Comment (edquist) [inline bosco-override-setup.sh:24]:
```suggestion
if [[ $GIT_ENDPOINT =~ ^[-A-Za-z0-9]+@ ]]; then
```

I believe hyphens are allowed in github usernames.  (Technically, not leading or trailing or multiple consecutive hyphens, but maybe it's not necessary to make the regex absolutely precise about that.)

Comment (edquist) [reply bosco-override-setup.sh:24]: Relatedly is this just for github repos or more generally for any git repo endpoint?

Comment (brianhlin) [reply bosco-override-setup.sh]: It's intended for any

**Comment by edquist** [comment]
> It's intended for any

I see, ok.  So the `if` check is basically to ensure that it's a git ssh url instead of a https:// or git://

Two notes about the general case:
1. unix usernames can contain underscores also (even though for github usernames this is not allowed)
2. a git ssh endpoint does not have to contain the username (eg, `m:pub/project.git` is a valid git ssh endpoint spec)

> I wish you could chain param expansions...

You can use capture groups for the regex though!

```
if [[ $GIT_ENDPOINT =~ ^[A-Za-z0-9_-]+@([^:]+): ]]; then
    ssh-keyscan "${BASH_REMATCH[1]}"
fi
```

I would say you could put the username part in an optional, non-capture group, like `(?:[A-Za-z0-9_-]+@)?`, in order to make the username optional ... but then it can match git:// and https:// urls, which i think you don't want.

**Pull request merged**